iModelCore/Units/tests/NonPublished/FormattingTests.cpp

```diff
@@ -1738,4 +1738,51 @@ TEST_F(FormattingTestFixture, AzimuthWithVariousBases) {
     ASSERT_STREQ("180.0°", formatAzimuth(90.0, 270.0,true).c_str());
 }
 
+TEST_F(FormattingTestFixture, DoubleToRatio){
+    auto formatDoubleToRatio = [](double value, Units::UnitCP unit, PresentationType type, DecimalPrecision precision = DecimalPrecision::Precision3) -> Utf8String
+        {
+            NumericFormatSpec integerRatioSpec;
+            integerRatioSpec.SetPresentationType(type);
+            integerRatioSpec.SetPrecision(precision);
+
+            Format integerRatioFormat(integerRatioSpec);
+
+            Units::Quantity quantity(value, *unit);
+            return integerRatioFormat.FormatQuantity(quantity);
+        };
+
+    {
+        auto verticalPerHorizontalUnit = s_unitsContext->LookupUnit("VERTICAL_PER_HORIZONTAL");
+
+        ASSERT_STREQ("1:0", formatDoubleToRatio(0.0, verticalPerHorizontalUnit, PresentationType::FractionalRatio).c_str()); // special case

```

Comment:
these calls could be easier to read if you define a lambda that takes a string and double input, and then reuse it for all these examples